### ISSUE TITLE
Field order when adding to MongoDB from 3rd party system.

### ISSUE THREAD

**Issue by JWDT** [doableware/djongo#122]
Not sure if I'm totally missing something obvious (fairly new to using MongoDB having grown up with SQL), but I'm trying to build a Django app that (at this stage) basically shows each user a list of messages.

There's a class for these messages, and I can use the Django admin system to add, edit, delete, etc. without any problems, until I add a "message" document directly to the MongoDB system directly (using Integromat, or Compass).

The document has the same fields, but in a different order, and that breaks it.

In some cases I get an AssertionError, usually when it tries to convert a field of one type to/from a different type (e.g. it thinks a CharField is actually a ArrayModelField).

In other cases it works, and shows me the data, but in the wrong order, so for the messages example it might look something like this (ignoring _id and other fields):

Database:

```
oid: 1
from: sender@example.com
to: receiver@example.com

oid: 2
to: sender@example.com
from: receiver@example.com

oid: 3
to: receiver@example.com
from: sender@example.com
```

Application:

ID: 1 (Correct)
From: sender@example.com
To: receiver@example.com

ID: 2 (Wrong)
From: sender@example.com
To: receiver@example.com

ID: 3 (Wrong)
From: receiver@example.com
To: sender@example.com

Basically it looks like it's using the order of the fields in the class and assuming MongoDB will always return documents in that order, totally ignoring the field name in the document. Is this expected, or have I done something wrong somewhere?

If it's expected behaviour, should it be? I thought it was quite normal for MongoDB to send documents with the fields in a different order for different documents.

**Comment by nesdis** [comment]
There is an issue with this, happens mostly after migrations are missed. Checking this.

**Comment by nesdis** [comment]
Latest commit should have fixed this

**Comment by JWDT** [comment]
Not sure if I tested this in 2018 but closing as presumably fixed or out of date.
